lib/rubocop/cop/lint/method_keyword_name.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Makes sure that methods does not reserver ruby keyword name.
+      #
+      # @example
+      #
+      #   # bad
+      #   def def
+      #     :ok
+      #   end
+      #
+      #   # bad
+      #   def false; end
+      #
+      class MethodKeywordName < Base
+        include AllowedIdentifiers
+
+        KEYWORDS = %i[
+          BEGIN END alias and begin break case class def defined?
+          do else elsif end ensure false for if in module next nil
+          not or redo rescue retry return self super then true undef
+          unless until when while yield
+        ].freeze
```

Comment:
A lot of these are defined in common use (not an exhaustive list):

* Ruby defines a `class` method on Kernel (https://ruby-doc.org/3.3.6/Kernel.html#method-i-class), `next` on `Enumerable`, `Date`, as an alias for `succ` on multiple classes, etc.
* RSpec defines an `and` method (https://www.rubydoc.info/github/rspec/rspec-expectations/RSpec/Matchers/Composable#and-instance_method)
* `begin` and `end` methods are defined in a lot of places, including `rubocop-ast`, `parser`, and ruby itself.